frontend/src/data/saga.ts

```diff
@@ -1,16 +1,16 @@
 import { all, call, put, takeLatest } from "redux-saga/effects";
 import FirebaseInteractor from "../firebase/firebaseInteractor";
+import { Action, ActionTypes, CreateUserParams, LoginParams, Word } from "../models/types";
 import {
   authenticationRequest,
   getWordsRequest,
-  singleRequest,
+  singleRequest
 } from "./actions";
-import {Action, ActionTypes, LoginParams, Word} from "../models/types";
 
 let firebaseInteractor = new FirebaseInteractor();
 
 export default function* rootSaga() {
-  yield all([root()]);
+  yield all([root(), waitForSignInAtStart()]);
```

Comment:
after deleting the single request in root, can you put waitForSignInAtStart() under root()? So like 

yield takeLatest(ActionTypes.WhateverRequest, waitForSignInAtStart);

Reply:
The only issue with that, is that then something needs to make the request (I think, not 100% sure if my mental model is correct). I did it in this style so that no matter where you start, you try to sign in. Does that make sense? I am not sure how to do it if I do it with the `takeLatest` style.

Reply:
hmmm kinda? cuz the thing about root() is that since it contains all the takeLatest then we can just do everything all at once. Would takeEvery solve that problem?